internal/xds/translator/route.go

```diff
@@ -15,22 +15,32 @@ import (
 	previoushost "github.com/envoyproxy/go-control-plane/envoy/extensions/retry/host/previous_hosts/v3"
 	matcherv3 "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
 	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/structpb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
 	"github.com/envoyproxy/gateway/internal/ir"
 	"github.com/envoyproxy/gateway/internal/utils/protocov"
 )
 
 const (
-	retryDefaultRetryOn             = "connect-failure,refused-stream,unavailable,cancelled,retriable-status-codes"
-	retryDefaultRetriableStatusCode = 503
-	retryDefaultNumRetries          = 2
+	retryDefaultRetryOn                 = "connect-failure,refused-stream,unavailable,cancelled,retriable-status-codes"
+	retryDefaultRetriableStatusCode     = 503
+	retryDefaultNumRetries              = 2
+	envoyGatewayMetadataNamespace       = "io.envoyproxy.gateway"
+	envoyGatewayMetadataKeyGroupVersion = "groupVersion"
+	envoyGatewayMetadataKeyKind         = "kind"
+	envoyGatewayMetadataKeyName         = "name"
+	envoyGatewayMetadataKeyNamespace    = "namespace"
+	envoyGatewayMetadataKeyAnnotations  = "annotations"
+	envoyGatewayMetadataKeySectionName  = "sectionName"
+	envoyGatewayMetadataKeyRoute        = "route"
```

Comment:
to which resource are we building this key off ? input gateway api, IR or xds filter ?

Reply:
will change to `envoyGatewayXds*` to make it clear that this is a const used in generated XDS.